cocos/audio/android/Track.h

```diff
@@ -0,0 +1,100 @@
+/****************************************************************************
+Copyright (c) 2016 Chukong Technologies Inc.
+
+http://www.cocos2d-x.org
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+****************************************************************************/
+#ifndef COCOS_TRACK_H
+#define COCOS_TRACK_H
+
+#include "audio/android/PcmData.h"
+#include "audio/android/IVolumeProvider.h"
+#include "audio/android/PcmBufferProvider.h"
+
+#include <functional>
+
+namespace cocos2d { namespace experimental {
+
+class Track : public PcmBufferProvider, public IVolumeProvider
+{
+public:
+    enum class State
+    {
+        IDLE,
+        PLAYING,
+        RESUMED,
+        PAUSED,
+        STOPPED,
+        OVER,
+        DESTROYED
+    };
+
+    Track(const PcmData &pcmData);
+    virtual ~Track();
+
+    inline State getState() const { return _state; };
+    inline void setState(State state)
+    {
+        if (_state != state)
+        {
+            _prevState = _state;
+            _state = state;
+            onStateChanged(_state);
+        }
+    };
+
+    inline State getPrevState() const { return _prevState; };
+
+    inline bool isPlayOver() const { return _state == State::PLAYING && mNextFrame >= mNumFrames;};
+    inline void setName(int name) { _name = name; };
```

Comment:
`Name` is something similar with `Texture id` stuff.
It's used by `AudioMixer` internally, and it has to be an integer.
You could refer to the API of `AudioMixer`:
```c++
    int         getTrackName(audio_channel_mask_t channelMask,
                             audio_format_t format, int sessionId);

    // Free an allocated track by name
    void        deleteTrackName(int name);

    // Enable or disable an allocated track by name
    void        enable(int name);
    void        disable(int name);

    void        setParameter(int name, int target, int param, void *value);

    void        setBufferProvider(int name, AudioBufferProvider* bufferProvider);
```